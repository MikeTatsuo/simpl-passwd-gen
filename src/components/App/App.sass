.App
  height: 100vh
  display: grid
  grid-template-columns: 15vw auto 15vw
  grid-template-rows: 56px auto 100px
  grid-template-areas: "header header header" "left main right" "footer footer footer"
