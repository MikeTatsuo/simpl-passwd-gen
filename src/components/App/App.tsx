import { Component } from 'react';

import Header from 'src/components/Header/Header';
import Card from 'src/components/Card/Card';

import { EnvConfig } from 'src/config';

import { CardFactoryTypes } from 'src/models';

import './App.sass';

const { title } = EnvConfig;
const { PASSWDGEN } = CardFactoryTypes;

export default class App extends Component {
	render() {
		return (
			<div className="App">
				<Header title={title} />
				<Card type={PASSWDGEN} />
			</div>
		);
	}
}
