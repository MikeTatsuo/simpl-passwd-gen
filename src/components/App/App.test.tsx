import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import App from './App';

let container: HTMLDivElement;

describe('App.tsx', () => {
	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		unmountComponentAtNode(container);
		container.remove();
	});

	describe('snapshot', () => {
		it('should match snapshot', () => {
			act(() => {
				render(<App />, container);
			});

			expect(container).toMatchSnapshot();
		});
	});
});
