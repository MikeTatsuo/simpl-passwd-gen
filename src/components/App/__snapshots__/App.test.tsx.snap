// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx snapshot should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="header"
    >
      <span
        class="header-title"
      >
        SimplPasswdGen
      </span>
    </header>
    <section
      class="card"
    >
      <section
        class="passwd-gen"
      >
        <p
          class="passwd-gen__title"
        >
          Password:
        </p>
        <p
          class="passwd-gen__result"
        />
        <section
          class="passwd-gen__options-container"
        >
          <p
            class="passwd-gen__subtitle"
          >
            Options:
          </p>
          <label
            for="passwd-gen__min-char"
          >
            Minimum Characters
          </label>
          <input
            class="passwd-gen__min-char"
            max="100"
            min="0"
            type="number"
            value="6"
          />
          <label
            for="passwd-gen__max-char"
          >
            Maximum Characters
          </label>
          <input
            class="passwd-gen__max-char"
            max="100"
            min="0"
            type="number"
            value="12"
          />
          <div
            class="checkbox"
          >
            <input
              checked=""
              class="checkbox__input"
              name="L"
              type="checkbox"
            />
            <label
              for="checkbox__label"
            >
              Lower case
            </label>
          </div>
          <div
            class="checkbox"
          >
            <input
              checked=""
              class="checkbox__input"
              name="U"
              type="checkbox"
            />
            <label
              for="checkbox__label"
            >
              Upper case
            </label>
          </div>
          <div
            class="checkbox"
          >
            <input
              checked=""
              class="checkbox__input"
              name="N"
              type="checkbox"
            />
            <label
              for="checkbox__label"
            >
              Number
            </label>
          </div>
          <div
            class="checkbox"
          >
            <input
              checked=""
              class="checkbox__input"
              name="S"
              type="checkbox"
            />
            <label
              for="checkbox__label"
            >
              Special Characters
            </label>
          </div>
        </section>
        <section
          class="passwd-gen__action-container"
        >
          <button>
            Clear
          </button>
          <button>
            Generate
          </button>
        </section>
      </section>
    </section>
  </div>
</div>
`;
